src/Microsoft.VisualStudio.Editors/ResourceEditor/Resource.vb

```diff
@@ -1630,11 +1630,11 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
         ''' file are only used to produce the .designer.cs file, which is still built against
         ''' netstandard.dll.
         ''' </remarks>
-        Private Shared Function AdjustSystemStringAssemblyQualifiedName(AssemblyQualifiedName As String) As String
-            Static SystemStringAssemblyQualifiedName As String
+        Private Shared Function AdjustAssemblyQualifiedName(AssemblyQualifiedName As String) As String
+            Static CorrectedAssemblyQualifiedName As String
 
             ' If this type definitely isn't System.String then bail out before we allocate
-            If Not AssemblyQualifiedName.StartsWith("System.String") Then
+            If Not (AssemblyQualifiedName.StartsWith("System.String") Or AssemblyQualifiedName.StartsWith("System.Drawing.Bitmap") Or AssemblyQualifiedName.StartsWith("System.Drawing.Icon")) Then
```

Comment:
Agree with Dave on this - If you this a map from old type name -> new type name this effectively just becomes a single lookup, and you don't need to keep this in sync with the if statements below.

Reply:
Consider ditching `StartsWith`. However unlikely, it does the wrong thing for a `System.String.Foo` type.

Probably easiest to parse out `typeName` first, then look it up in a dictionary as mentioned above.

Try to avoid a code path that doesn't explicitly return anything. If the input string was `"Foo"`, this method would currently return `Nothing`. It should probably return `AssemblyQualifiedName` in cases where no change is needed.